std/experimental/allocator/gc_allocator.d

```diff
@@ -197,3 +197,146 @@ pure nothrow @safe unittest
     void *badPtr = (() @trusted => cast(void*)(0xdeadbeef))();
     assert(GCAllocator.instance.resolveInternalPointer(badPtr, result) == Ternary.no);
 }
+
+/**
+Manual allocator that allocates from the D runtime's garbage-collected heap.
+*/
+struct GCHeapMallocator
+{
+    import std.typecons : Ternary;
+
+    private GCAllocator impl;
+
+    /**
+    The alignment is a static constant equal to `platformAlignment`, which
+    ensures proper alignment for any D data type.
+    */
+    enum uint alignment = impl.alignment;
+
+    /**
+    Standard allocator methods per the semantics defined above. The $(D
+    deallocate) and `reallocate` methods are `@system` because they may
+    move memory around, leaving dangling pointers in user code.
+    */
+    pure nothrow @trusted void[] allocate(size_t bytes) shared const
+    {
+        return impl.allocate(bytes);
+    }
+
```

Comment:
Would `alias this` work here, instead of explicit forwards?

Reply:
`alias this` to a `private` field doesn't work when used from outside the module, and even if it did, I don't think the implicit conversion would be desirable.